Source/WebCore/Modules/indexeddb/server/SQLiteIDBBackingStore.cpp

```diff
@@ -2709,6 +2714,21 @@ void SQLiteIDBBackingStore::forEachObjectStoreRecord(const IDBResourceIdentifier
     }
 }
 
+void SQLiteIDBBackingStore::incrementalVacuumIfNeeded()
+{
+    ASSERT(m_sqliteDB);
+    ASSERT(m_sqliteDB->isOpen());
+
+    const auto freeSpaceSize = m_sqliteDB->freeSpaceSize();
+    const auto totalSize = m_sqliteDB->totalSize();
+
+    if (totalSize <= 10 * freeSpaceSize) {
```

Comment:
Why not just do vaccum unconditionally? (Curious how we get the number 10)

Reply:
I think it bases how it was with web sql database: https://github.com/WebKit/WebKit/blob/main/Source/WebCore/Modules/webdatabase/Database.cpp#L744